### PR TITLE
Implementation of Bunyan Logger

### PR DESCRIPTION
**Pull request by haeferer**
Hi, 

this PR implements the bunyan logger the same way as the last try (it's the same code)

known problems:

flow:
- Type error between defaultsettings from spark-protocol and spark-server, because they are merged. This should go away if spark-protocol also uses bunyan or at least the same LOG_LEVELS

### CODE REVIEW

File: examples/eventProvider.js [hunk 1]
```diff
@@ -6,15 +6,44 @@
...
 
 const container = new Container();
 defaultBindings(container, settings);
 
-container.constitute('EVENT_PROVIDER').onNewEvent((event: Event) => {
-  // do piping stuff here.
+const deviceServer = container.constitute('DeviceServer');
+deviceServer.start();
+
+/*
```
Comment (jlkalberer) [inline examples/eventProvider.js]: Delete this. We try to never check in commented out code.

File: src/main.js [hunk 2]
```diff
@@ -70,13 +73,13 @@ const addresses = arrayFlatten(
     // eslint-disable-next-line no-unused-vars
     ([name, nic]: [string, mixed]): Array<string> =>
       (nic: any)
-        .filter(
-          (address: Object): boolean =>
-            address.family === 'IPv4' && address.address !== '127.0.0.1',
-        )
+        .filter((address: Object): boolean => {
+          logger.debug({ found: address }, 'Network Interface');
```
Comment (jlkalberer) [inline src/main.js]: Why are you logging this here? All these addresses will be logged on line 84

File: examples/eventProvider.js (hunk 1 above)
Comment (haeferer) [inline examples/eventProvider.js]: Done

File: src/main.js (hunk 2 above)
Comment (haeferer) [inline src/main.js]: Based on my expceriences in virtual machines and docker, the information which network interfaces are available is sometime usefull, especially if you select one to bind the server to.

But i will remove this log, cause this is only informational (and is not! used to bind the server to this interface) 
-> I will remove the info.
 
done

**Comment by haeferer** [comment]
With PR in spark-protocol you mean

* add types for LOG_LEVEL to remove flow-type error
or

* implement complete bunyan in sparc-protocol?

File: examples/eventProvider.js [hunk 3]
```diff
@@ -6,15 +6,42 @@
...
 
 const container = new Container();
 defaultBindings(container, settings);
 
-container.constitute('EVENT_PROVIDER').onNewEvent((event: Event) => {
-  // do piping stuff here.
+const deviceServer = container.constitute('DeviceServer');
+deviceServer.start();
+
+const evProvider = container.constitute('EVENT_PROVIDER');
```
Comment (AntonPuko) [inline examples/eventProvider.js]: nit: eventProvider

File: examples/eventProvider.js [hunk 4]
```diff
@@ -6,15 +6,42 @@
...
+evProvider.onNewEvent((event: Event) => {
+  logger.info('Event onNewEvent',event);
+});
+
+
+
+deviceServer._eventPublisher.subscribe('*', (ev: any) => {
```
Comment (AntonPuko) [inline examples/eventProvider.js]: remove this

File: src/app.js [hunk 5]
```diff
@@ -37,22 +39,32 @@ export default (
       });
       return response.sendStatus(204);
     }
-    response.set({ 'Access-Control-Allow-Origin': '*' });
+    response.set({
```
Comment (AntonPuko) [inline src/app.js]: is this perttier output?

File: src/lib/WebhookLogger.js [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+import Logger from './logger';
+const logger = Logger.createModuleLogger(module);
 
 class WebhookLogger implements IWebhookLogger {
   _lastLog: Array<any>;
 
-  log(...args: Array<any>) {
-    this._lastLog = args;
-    logger.log(...args);
+  log(...argsarray: Array<any>) {
```
Comment (AntonPuko) [inline src/lib/WebhookLogger.js]: nits: args

File: src/lib/WebhookLogger.js [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+    this._lastLog = argsarray;
+    logger.info({ args: argsarray }, 'WebHook');
```
Comment (AntonPuko) [inline src/lib/WebhookLogger.js]: nit: webhook

File: src/app.js (hunk 5 above)
Comment (haeferer) [inline src/app.js]: from last run

File: src/managers/WebhookManager.js [hunk 8]
```diff
@@ -27,6 +28,7 @@ const parseEventData = (event: Event): Object => {
     }
     return {};
   } catch (error) {
+    logger.warn({ err: error, evdata: event.data }, 'parseEventData failed');
```
Comment (AntonPuko) [inline src/managers/WebhookManager.js]: nit: eventData

File: examples/eventProvider.js (hunk 3 above)
Comment (haeferer) [inline examples/eventProvider.js]: done

File: examples/eventProvider.js (hunk 4 above)
Comment (haeferer) [inline examples/eventProvider.js]: Done

File: src/lib/WebhookLogger.js (hunk 6 above)
Comment (haeferer) [inline src/lib/WebhookLogger.js]: done

File: src/lib/WebhookLogger.js (hunk 7 above)
Comment (haeferer) [inline src/lib/WebhookLogger.js]: I dont event know when this is called. The text is original from previous code. 
I renamed to "WebHookLogger called"

File: src/managers/WebhookManager.js (hunk 8 above)
Comment (haeferer) [inline src/managers/WebhookManager.js]: done

File: src/app.js (hunk 5 above)
Comment (haeferer) [inline src/app.js]: i run prettifier again. nothing changes. The multil-ines looks much prettier for me

File: src/lib/WebhookLogger.js (hunk 7 above)
Comment (AntonPuko) [inline src/lib/WebhookLogger.js]: just rename `'WebHook'` to `webhook`

**Comment by haeferer** [comment]
Finished.

**Comment by AntonPuko** [comment]
looking ok, so we'll merge it after bunyan pr for spark-protocol.

**Comment by jlkalberer** [comment]
Thanks @haeferer

**Comment by snazzypants** [comment]
Installed from scratch today...ran into this error when starting the server:

```
Error: Cannot find module 'bunyan'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/root/spark-server/src/lib/logger.js:22:1)
    at Module._compile (module.js:570:32)
    at loader (/root/spark-server/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/root/spark-server/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)

```

**Comment by haeferer** [comment]
I will look into this. 
* Which plattform do you use? 
* How do you installed from scratch?  ```npm install``` (any errors?)
* Did you also install the bunyan-cli (```npm install bunyan -g```). It should work without the tool

**Comment by snazzypants** [comment]
Thanks. I had spun up a fresh server running NodeJS 6.9.5 on Ubuntu 14.04, cloned the project, and ran `npm install`.  No errors on install.

I did end up installing bunyan separately, just to see what would happen. It then complained that bunyan-middleware was missing.  I found that odd since the dependencies were listed in package.json.

PS. I saw your other issue about running npm install twice (don't think I ever did that).  Anyway, I had the exact same error when trying to run the server from /dist.

**Comment by haeferer** [comment]
There is a bug on blank installations. I will look into this the next hours.
The bug is based on the missing /third-parties/ directory.
This is generated during install (the lines with 'downloading...'). This directory can be recreated by npm run update-firmware.
Problem: the directory is generated in the wrong place :(

Please note: There are problems with (the latest) NPM. Which version of NPM do you use

**Pull request merged**